File: api/mocks/pokemon.go
package mocks

import "github.com/WebWizardsDev/poke-db/api/models"

var MockPokemonSpeciesBulbasaur = models.PokemonSpecies{
	ID:                   1,
	HasGenderDifferences: false,
	HatchCounter:         20,
	IsBaby:               false,
	IsLegendary:          false,
	IsMythical:           false,
	Name:                 "bulbasaur",
	Varieties:            MockPokemonVarietiesBulbasaur,
}

var MockPrimaryTypeId uint = 1
var MockPSecondaryTypeId uint = 2

var MockPokemonVarietiesBulbasaur = []models.Pokemon{{
	ID:               1,
	Name:             "bulbasaur",
	PrimaryType:      &MockTypesGrass,
	PrimaryTypeId:    &MockPrimaryTypeId,
	SecondaryType:    &MockTypesPoison,
	SecondaryTypeId:  &MockPSecondaryTypeId,
	PokemonSpeciesId: 1,
	SpriteUrl:        "https://example.com/pokemon.png",
	Cry:              "https://example.com/pokemon.ogg",
	Weight:           20,
	Height:           5,
	IsDefault:        false,
}}

var MockTypesGrass = models.MoveType{}
var MockTypesPoison = models.MoveType{}

var MockListPokemonResponse = models.ListPokemonResponse{
	Pokemon:  []models.PokemonSpecies{MockPokemonSpeciesBulbasaur},
	Page:     1,
	PageSize: 100,
	Total:    1,
	Params:   models.ListPokemonParams{},
	NextPage: "",
	PrevPage: "",
}

File: frontend/src/query/api/pokemon.ts
import axios from "axios";
import {ListPokemonResponse} from "@/models/pokemon.ts";

export const getPokemon = async (page: number, pokemonName?: string, pokemonTypeId?: string) => {
    const {data} = await axios.get<ListPokemonResponse>(`/api/v1/pokemon`, {
        params: {
            page,
            pokemonName,
            pokemonTypeId,
            pageSize: 12
        }
    });

    return {
        pokemon: data.pokemon,
        total: data.total,
        pageSize: data.pageSize,
        pokemonName: data.params.pokemonName,
        pokemonTypeId: data.params.pokemonTypeId
    };
}
